html/user/create_account.php

```diff
@@ -91,6 +95,25 @@
     }
 }
 
+// If the project has configured itself to record the opt-in consent
+// time, then do so here. Defaults to setting consent_id to '1' - the
+// general opt-in consent type.
+if (parse_bool($config, "enable_record_optin_consent")) {
+    if ( (!is_null($optin)) and $source) {
+        // Record the user giving consent in database - if optin is 0,
+        // this is an anonymous account and consent_not_required is
+        // set to 1.
+        if ($optin==0) {
+            $rc = consent_to_termsofuse($user, 1, 0, 1, $source);
+        } else  {
+            $rc = consent_to_termsofuse($user, 1, 1, 0, $source);
+        }
+        if (!$rc) {
+            xml_error(-1, "database error, please contact site administrators");
+        }
+    }
```

Comment:
If consent is required but whoever is creating the account knows nothing of consent then what? I think the code that checks that consent was given should go before `make_user` call.

Reply:
Right now there are multiple ways this Web RPC is called, including simply typing in `https://project_url/create_account.php?email_addr=xyz@example.com&passwd_hash=ahash&username=name`

This means anyone can create an account without knowledge of a project's terms of use/opt-in consent. The boinc-command line client is another place where you can create an account without ever seeing the terms of use and consenting to it.

Ultimately, projects which are in Europe and really care about GDPR should actually turn off this create_account RPC and only allow users to register via their Web site, which guarantees the user sees the terms of use.

Reply:
I was thinking that Manager and boinccmd would be later enhanced to pass `optin` and `source` parameters. A user could be using some third party manager that doesn't show TOU and the user might not know he needs to give consent.

I was under the impression that new (non-anonymous) accounts can't be created unless the user consents to TOU. But that raises the question, if user has not given or revoked consent to TOU at all, then does that make the account automatically anonymous?